src/server/string_family.cc

```diff
@@ -413,67 +412,78 @@ OpResult<array<int64_t, 5>> OpThrottle(const OpArgs& op_args, const string_view
     if (!opt_prev) {
       return OpStatus::INVALID_VALUE;
     }
-    tat_ms = *opt_prev;
+    tat_ns = *opt_prev;
   }
 
-  int64_t new_tat_ms = max(tat_ms, now_ms);
-  if (new_tat_ms > INT64_MAX - increment_ms) {
+  int64_t new_tat_ns = max(tat_ns, now_ns);
+  if (new_tat_ns > INT64_MAX - increment_ns) {
     return OpStatus::INVALID_INT;
   }
-  new_tat_ms += increment_ms;
+  new_tat_ns += increment_ns;
 
-  if (new_tat_ms < INT64_MIN + delay_variation_tolerance_ms) {
+  if (new_tat_ns < INT64_MIN + delay_variation_tolerance_ns) {
     return OpStatus::INVALID_INT;
   }
-  const int64_t allow_at_ms = new_tat_ms - delay_variation_tolerance_ms;
 
-  if (allow_at_ms >= 0 ? now_ms < INT64_MIN + allow_at_ms : now_ms > INT64_MAX + allow_at_ms) {
+  // The cutoff point before which a request is rejected (throttled) and at or after which a request
+  // is accepted.
+  const int64_t allow_at_ns = new_tat_ns - delay_variation_tolerance_ns;
+
+  if (allow_at_ns >= 0 ? now_ns < INT64_MIN + allow_at_ns : now_ns > INT64_MAX + allow_at_ns) {
```

Comment:
please factor out this into a separate variable

Reply:
added a function for this kind of check